docs/migrating-from-terrajet-to-upjet.md

```diff
@@ -0,0 +1,106 @@
+For an example invocation, please see
+[here](https://github.com/upbound/upjet-provider-template/blob/d34119409586f6205ec8ed4b9b2c2481c74bf07e/config/provider.go#L29).
+- In `config/provider.go`:
+[pkg/config.WithDefaultResourceFn](https://github.com/crossplane/terrajet/blob/8d0ed485f9511b65a8f3a83801092bcae60678dd/pkg/config/provider.go#L143)
+has been removed from Upjet. You may instead use
+[pkg/config.WithDefaultResourceOptions](https://github.com/upbound/upjet/blob/c82119f5ef342f752406a0ed38264940b02e795f/pkg/config/provider.go#L154).
+For an example invocation, please see
+[here](https://github.com/upbound/upjet-provider-template/blob/d34119409586f6205ec8ed4b9b2c2481c74bf07e/config/provider.go#L31).
+- In `config/provider.go`: If your provider does not have "base" API packages
+(like the `ProviderConfig` packages) at both `v1alpha1` and `v1beta1` versions,
+then you may need to override the provider's [`BasePackages`
+configuration](https://github.com/upbound/upjet/blob/7e84c638a8bc5c93c6da3cf9420f961f165dd05d/pkg/config/provider.go#L76)
+configuration using
+[pkg/config.WithBasePackages](https://github.com/upbound/upjet/blob/7e84c638a8bc5c93c6da3cf9420f961f165dd05d/pkg/config/provider.go#L146).
+This is because Upjet
+[assumes](https://github.com/upbound/upjet/blob/7e84c638a8bc5c93c6da3cf9420f961f165dd05d/pkg/config/common.go#L18)
+both `apis/v1alpha1` `apis/v1beta1` base API packages and the
+`internal/controller/providerconfig` controller package exist for your provider.
+If, for example, only the `apis/v1alpha1` base API package is available for your
+provider, then you may configure your provider with:
+```go
+config.WithBasePackages(config.BasePackages{
+			APIVersion: []string{
+				// Default package for ProviderConfig APIs
+				"apis/v1alpha1",
+			},
+			Controller: []string{
+				// Default package for ProviderConfig controllers
+				"internal/controller/providerconfig",
+			},
+		})
+```
+as a
+[config.ProviderOption](https://github.com/upbound/upjet/blob/7e84c638a8bc5c93c6da3cf9420f961f165dd05d/pkg/config/provider.go#L115)
+passed to
+[pkg/config.NewProvider](https://github.com/upbound/upjet/blob/c82119f5ef342f752406a0ed38264940b02e795f/pkg/config/provider.go#L172).
+
+5. When you try to build the provider package, you may also encounter some
+compilation errors regarding managed resource configurations:
+- In various `apis/<resource API group>/<API
+version>/zz_generated.resolvers.go`: You may observe errors referring to
+undeclared `spec` struct fields with names containing acronyms such as:
+```shell
+apis/ec2/v1alpha2/zz_generated.resolvers.go:195:38: mg.Spec.ForProvider.SecurityGroupIdRefs undefined (type InstanceParameters has no field or method SecurityGroupIdRefs)
+```
+You will also need to fix the referencer configurations for such managed
+resources using the [resource configuration API].
```

Comment:
```suggestion
You will also need to fix the referenced configurations for such managed
resources using the [resource configuration API].
```

Reply:
Replaced with "reference configurations" and added a link.